### PR TITLE
Add block for embedding YouTube live chat

### PR DESCRIPTION
**Pull request by iandunn**
This will help organizers who want to use YouTube as their streaming platform.

Testing is pretty straight-forward. `youtube-live-chat.js` has some notes at the top about examples of good URLs to test with.

Screen Shot 2020-04-15 at 17 46 08

### CODE REVIEW

File: public_html/wp-content/mu-plugins/virtual-embeds/virtual-embeds.php [hunk 1]
```diff
@@ -78,3 +89,64 @@ function render_crowdcast_block( $attributes ) {
...
+		 * that in passing in their documentation.
+		 *
+		 * There don't seem to be any practical workarounds. See https://stackoverflow.com/a/12845483/450127
+		 * and https://stackoverflow.com/a/58739531/450127.
+		 *
+		 * So... yeah. We're left with this.
+		 */
+		?>
+
+		<p class="wp-block-wordcamp-youtube-live-chat-embed__boneheaded-decision">
```
Comment (ryelle) [inline public_html/wp-content/mu-plugins/virtual-embeds/virtual-embeds.php]: I'm not really comfortable with this being in our public-facing code, even if it is a very strange thing for youtube to do. What about this?

```suggestion
		<p class="wp-block-wordcamp-youtube-live-chat-embed__availability-warning">
```

File: public_html/wp-content/plugins/wc-post-types/package.json [hunk 2]
```diff
@@ -22,13 +22,14 @@
 		"lint:js": "wp-scripts lint-js js/src",
 		"lint:pkg-json": "wp-scripts lint-pkg-json"
 	},
+	"eslintConfig:comment": {
+		"root": "Setting this to true is necessary to avoid merging WordCamp.org configs with any that exist in parent folders in the developer's environment. See https://github.com/eslint/eslint/commit/c99e9b621fe46b8535cf59fe489977dd36cb2a39."
+	},
 	"eslintConfig": {
-		"extends": "plugin:@wordpress/eslint-plugin/recommended",
+		"root": true,
+		"extends": "../../mu-plugins/blocks/.eslintrc.js",
```
Comment (ryelle) [inline public_html/wp-content/plugins/wc-post-types/package.json:30]: This is unrelated to the PR — can you take this out before merging?

If we want to share one customized eslint config across the whole project, we should make that the project root, not pull in from a separate project like this.

**Review by ryelle — COMMENTED** [review]
2 comments (that I would like acknowledged or fixed before merging), otherwise it's testing well 👍

File: public_html/wp-content/mu-plugins/virtual-embeds/virtual-embeds.php (hunk 1 above)
Comment (iandunn) [reply public_html/wp-content/mu-plugins/virtual-embeds/virtual-embeds.php]: It seems pretty mild to me, and something they deserve to be publicly criticized for, but I don't feel strongly. I'll update the CSS reference too.

File: public_html/wp-content/plugins/wc-post-types/package.json (hunk 2 above)
Comment (iandunn) [reply public_html/wp-content/plugins/wc-post-types/package.json:30]: I don't see any reason for us to have inconsistent standards within the same codebase. I agree that it'd be best to have 1 file at the root, but that would be ever farther out of scope. Doing this much was necessary for this PR since it directly affected the new code this PR adds.

This moves us closer to having 1 standard, and is very easy to implement. We can always iterate later to move to a single file.

Since the block is time-sensitive, I'll go ahead and merge the other commits, and leave this one just in this branch, so that we can continue discussing delaying the approved commits.

Comment (ryelle) [reply public_html/wp-content/plugins/wc-post-types/package.json:30]: I agree we should stick to one eslint config, and if you push this commit to another PR we can move forward.

> Since the block is time-sensitive, I'll go ahead and merge the other commits, and leave this one just in this branch, so that we can continue discussing delaying the approved commits.

That's exactly what I was suggesting.

Comment (ryelle) [reply public_html/wp-content/plugins/wc-post-types/package.json:30]: Well, I see after I commented that … this isn't _exactly_ what I meant. Now that you've merged those commits, this is full of conflicts. Now you can close this branch, create a new branch off production, cherry pick this commit, and open a new PR to make this change 👍

Comment (ryelle) [reply public_html/wp-content/plugins/wc-post-types/package.json:30]: BTW, i'm happy to kick that off if you're busy with more urgent things.

Comment (iandunn) [reply public_html/wp-content/plugins/wc-post-types/package.json:30]: That would be great, thank you!

Comment (iandunn) [reply public_html/wp-content/plugins/wc-post-types/package.json:30]: Rebase resolved the conflicts.

**Comment by ryelle** [comment]
Closing this, linter config work will continue in #438
